### ISSUE TITLE
Custom template scripts

### ISSUE THREAD

**Issue by tehdirector** [grafana/grafana#1215]
Is there any interest in having the ability to call an outside script (python, perl, bash) as the template query to pull in dynamically created variables? 

I have a number of use cases were I would like to populate the template variables with dynamic content that right now I have to hard code in as a custom template variable and modify if/when the list changes. Having the ability to call a script that I can create to return a comma delimited list of variables would be amazing for me, just not sure if anyone else could benefit from this. To fit my use case exactly I would envision it being able to pass arguments such as other defined variable with the script...

Say I have variables set as follows:

$var1 = site.*
$var2 = site.$var1.something.*
$var3 = mycustomscript.py $var2

This would allow me to create a script that would pull a custom list from my applications using the variables i've already set within my templates. I define my site which determines the list of variables available in my second variable and in turn creates a list of values for my third variable.

I hope this is something that might be possible as well as useful for others.

**Comment by jjneely** [comment]
I came across this as well.  We've got our machines setup into environments, QA2, QA1, PROD, etc which are all easily found by inspecting the metrics in Graphite.  However, our Canary machines which get deployed to first, are a small subset of PROD.  I'm looking for a way to Grafana to be able to select these machines with a template variable.

Our CMDB knows which machines are Canary machines, but that detail is not in our graphite metric tree.
